tests/core/pyspec/eth2spec/merge/mainnet.py

```diff
@@ -0,0 +1,2931 @@
+    _get_base_reward, lru_size=2048)
+
+_get_committee_count_per_slot = get_committee_count_per_slot
+get_committee_count_per_slot = cache_this(
+    lambda state, epoch: (state.validators.hash_tree_root(), epoch),
+    _get_committee_count_per_slot, lru_size=SLOTS_PER_EPOCH * 3)
+
+_get_active_validator_indices = get_active_validator_indices
+get_active_validator_indices = cache_this(
+    lambda state, epoch: (state.validators.hash_tree_root(), epoch),
+    _get_active_validator_indices, lru_size=3)
+
+_get_beacon_committee = get_beacon_committee
+get_beacon_committee = cache_this(
+    lambda state, slot, index: (state.validators.hash_tree_root(), state.randao_mixes.hash_tree_root(), slot, index),
+    _get_beacon_committee, lru_size=SLOTS_PER_EPOCH * MAX_COMMITTEES_PER_SLOT * 3)
+
+_get_matching_target_attestations = get_matching_target_attestations
+get_matching_target_attestations = cache_this(
+    lambda state, epoch: (state.hash_tree_root(), epoch),
+    _get_matching_target_attestations, lru_size=10)
+
+_get_matching_head_attestations = get_matching_head_attestations
+get_matching_head_attestations = cache_this(
+    lambda state, epoch: (state.hash_tree_root(), epoch),
+    _get_matching_head_attestations, lru_size=10)
+
+_get_attesting_indices = get_attesting_indices
+get_attesting_indices = cache_this(
+    lambda state, data, bits: (
+        state.randao_mixes.hash_tree_root(),
+        state.validators.hash_tree_root(), data.hash_tree_root(), bits.hash_tree_root()
+    ),
+    _get_attesting_indices, lru_size=SLOTS_PER_EPOCH * MAX_COMMITTEES_PER_SLOT * 3)
+
+
+def get_generalized_index(ssz_class: Any, *path: Sequence[PyUnion[int, SSZVariableName]]) -> GeneralizedIndex:
+    ssz_path = Path(ssz_class)
+    for item in path:
+        ssz_path = ssz_path / item
+    return GeneralizedIndex(ssz_path.gindex())
+
+
+ExecutionState = Any
+
+
+def get_pow_block(hash: Bytes32) -> PowBlock:
+    return PowBlock(block_hash=hash, parent_hash=Bytes32(), is_valid=True, is_processed=True,
+                    total_difficulty=uint256(0), difficulty=uint256(0))
+
+
+def get_execution_state(execution_state_root: Bytes32) -> ExecutionState:
+    pass
+
+
+def get_pow_chain_head() -> PowBlock:
+    pass
+
+
+class NoopExecutionEngine(ExecutionEngine):
+
+    def on_payload(self, execution_payload: ExecutionPayload) -> bool:
+        return True
+
+    def set_head(self, block_hash: Hash32) -> bool:
+        return True
+
+    def finalize_block(self, block_hash: Hash32) -> bool:
+        return True
+
+    def assemble_block(self, block_hash: Hash32, timestamp: uint64, random: Bytes32) -> ExecutionPayload:
+        raise NotImplementedError("no default block production")
+
+
+EXECUTION_ENGINE = NoopExecutionEngine()
+
+
+assert FINALIZED_ROOT_INDEX == get_generalized_index(BeaconState, 'finalized_checkpoint', 'root')
+assert NEXT_SYNC_COMMITTEE_INDEX == get_generalized_index(BeaconState, 'next_sync_committee')
```

Comment:
I don't think you want to commit this files. @hwwhww aren't this gitignored or what may have happened?

Reply:
Right, it should be ignored. The spec name stub "merge" was very outdated. I guess @nalinbhardwaj didn't clean the old stub folder before pulling from the `dev` branch.